source-academy/cadet-frontend
# GitHub OAuth Login

### Description
Adds login capability for GitHub integration through the OAuth workflow.

I had to install 'react-simple-keyboard' in order for the project to be able to run after pulling from the latest changes. This is the cause of the huge increase in lines.

### Type of change

### How to test
1. Open the .env file and set REACT_APP_GITHUB_CLIENT_ID=d376715d50d04bd54a7e
2. Deploy the project.
3. Open the GitHub control buttons menu on the playground and press 'Log In'. Approve the app in the pop-up.
4. After the login has succeeded, the 'Log In' option will be replaced with a 'Log Out' option. The other options will also become available, but their functionality has not been implemented.
5. Click on the 'Log Out' option. The other options should now be greyed out while 'Log Out' is replaced by 'Log In' again.

### Checklist
